lib/cdo/poste.rb

```diff
@@ -209,29 +209,28 @@ def self.attachment_dir
     path
   end
 
-  # Takes a hash of name=>content, saves each to a file, and returns a
-  # hash of name=>saved_filename
+  # Takes a hash of name=>content, saves each to S3, and returns a
+  # hash of name=>saved_s3_key
   def self.save_attachments(attachments)
-    # Prevent saving attachments on a non-daemon server,
-    # to avoid missing file errors in deliver_poste_messages which runs on the daemon
-    raise "Emails with attachments can only be generated on a daemon server" unless CDO.daemon
-
     timestamp = DateTime.now.strftime('%Y%m%d_%H%M_%S%L')
     {}.tap do |saved|
       attachments.each do |name, content|
-        filename = File.expand_path "#{attachment_dir}/#{timestamp}-#{name}"
-        File.open(filename, 'w+b') {|f| f.write content}
-        saved[name] = filename
+        filename = "#{timestamp}-#{name}"
```

Comment:
I'd recommend having a directory per environment within the bucket - we do this for some of our other buckets such as assets and files. Will allow us to apply more restrictive AWS permissions for each environment in the future, and it could be useful for auditability purposes in production as well.